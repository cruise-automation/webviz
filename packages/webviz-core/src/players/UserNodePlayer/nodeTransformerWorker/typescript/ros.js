// @flow
//
//  Copyright (c) 2019-present, Cruise LLC
//
//  This source code is licensed under the Apache License, Version 2.0,
//  found in the LICENSE file in the root directory of this source tree.
//  You may not use this file except in compliance with the License.
export const ros_lib_filename = "ros/index.d.ts";
export const ros_lib_dts = `
  // Placeholder module declaration. Once any bag-like data source is added to
  // Webviz, this module will be autogenerated.
  declare module "ros";
`;
