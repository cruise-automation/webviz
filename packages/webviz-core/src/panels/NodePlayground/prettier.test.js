// @flow
//
//  Copyright (c) 2020-present, Cruise LLC
//
//  This source code is licensed under the Apache License, Version 2.0,
//  found in the LICENSE file in the root directory of this source tree.
//  You may not use this file except in compliance with the License.

import getPrettifiedCode from "./prettier";

describe("getPrettifiedCode", () => {
  it("formats valid Typescript code", async () => {
    const unformattedValidCode = `
      import { Input,   Messages  } from "ros";;
      const VALUE =
      "val";
      const publisher = (message: Input< "/foo/bar">
      ): Messages.visualization_msgs__WebvizMarkerArray | undefined => {
        return { VALUE
        }}
    `;
    const formattedCode = await getPrettifiedCode(unformattedValidCode);
    expect(formattedCode).toMatchSnapshot();
  });

  it("throws an error for invalid code", async () => {
    const unformattedInvalidCode = `
      import { Input,  from "ros"; // Missing closing curly
      const publisher = (): => { // Missing type
        return;
      }
    `;
    expect(getPrettifiedCode(unformattedInvalidCode)).rejects.not.toBeFalsy();
  });
});
