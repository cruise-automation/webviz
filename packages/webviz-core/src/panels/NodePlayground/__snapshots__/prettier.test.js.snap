// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getPrettifiedCode formats valid Typescript code 1`] = `
"import { Input, Messages } from \\"ros\\";
const VALUE = \\"val\\";
const publisher = (
  message: Input<\\"/foo/bar\\">
): Messages.visualization_msgs__WebvizMarkerArray | undefined => {
  return { VALUE };
};
"
`;
